File: app/src/main/java/ru/gb/lesson12/fragment/ListFragment.java
package ru.gb.lesson12.fragment;

import android.content.Context;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.ItemTouchHelper;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import ru.gb.lesson12.R;
import ru.gb.lesson12.data.InMemoryRepoImpl;
import ru.gb.lesson12.data.Note;
import ru.gb.lesson12.data.PopupMenuClick;
import ru.gb.lesson12.data.Repo;
import ru.gb.lesson12.recycler.NoteHolder;
import ru.gb.lesson12.recycler.NotesAdapter;

public class ListFragment extends Fragment

{

    private Repo repository = InMemoryRepoImpl.getInstance();
    private NotesAdapter adapter = new NotesAdapter();
    RecyclerView listAdapter;
    private Note note;

    public interface RecyclerController {
        void connect();
    }

    private RecyclerController recyclerController;

    @Override
    public void onAttach(@NonNull Context context) {
        this.recyclerController = (RecyclerController) context;
        super.onAttach(context);
    }

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {

        return inflater.inflate(R.layout.fragment_recycle_note, container, false);
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {

        adapter.setOnPopupMenuClick((PopupMenuClick) requireContext());
        listAdapter = view.findViewById(R.id.list_notes);
        listAdapter.setAdapter(adapter);
        listAdapter.setLayoutManager(new LinearLayoutManager(requireContext()));

        adapter.setNotes(repository.getAll());

        // Helper
        ItemTouchHelper helper = new ItemTouchHelper(new ItemTouchHelper.Callback() {

            @Override
            public int getMovementFlags(@NonNull RecyclerView recyclerView, @NonNull RecyclerView.ViewHolder viewHolder) {
                int swipeFlag = ItemTouchHelper.START | ItemTouchHelper.END;
                return makeMovementFlags(0, swipeFlag);
            }

            @Override
            public boolean onMove(@NonNull RecyclerView recyclerView, @NonNull RecyclerView.ViewHolder viewHolder, @NonNull RecyclerView.ViewHolder target) {
                return false;
            }

            @Override
            public void onSwiped(@NonNull RecyclerView.ViewHolder viewHolder, int direction) {
                int position = viewHolder.getAdapterPosition();
                NoteHolder holder = (NoteHolder) viewHolder;
                Note note = holder.getNote();
                repository.delete(note.getId());
                adapter.delete(repository.getAll(), position);

            }
        });
        helper.attachToRecyclerView(listAdapter);
    }

    public Note passData(Note note) {
        this.note = note;
        return note;
    }

    public void delete(Note note, int position) {

        repository.delete(note.getId());
        adapter.delete(repository.getAll(), position);
    }

    public void update(Note note){
        repository.update(note);
        adapter.setNotes(repository.getAll());
    }

    public void create(Note note) {
        repository.create(note);
        adapter.setNotes(repository.getAll());
    }
}

File: app/src/main/java/ru/gb/notes/ui/fragment/ListFragment.kt
package ru.gb.notes.ui.fragment

import android.content.Context
import android.content.SharedPreferences
import android.os.Bundle
import android.preference.PreferenceManager
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.recyclerview.widget.ItemTouchHelper
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import ru.gb.notes.R
import ru.gb.notes.data.InMemoryRepoImpl
import ru.gb.notes.domain.Note
import ru.gb.notes.data.PopupMenuClick
import ru.gb.notes.ui.recycler.NoteHolder
import ru.gb.notes.ui.recycler.NotesAdapter

class ListFragment : Fragment() {
    private val repository = InMemoryRepoImpl.instance
    private val adapter = NotesAdapter()
    private lateinit var listAdapter: RecyclerView
    private var note: Note? = null
    private var prefs: SharedPreferences? = null

    interface RecyclerController {
        fun connect()
    }

    private var recyclerController: RecyclerController? = null
    override fun onAttach(context: Context) {
        recyclerController = context as RecyclerController
        prefs = PreferenceManager.getDefaultSharedPreferences(context)
//        prefs = context.getSharedPreferences(context)

        super.onAttach(context)
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
    }

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        return inflater.inflate(R.layout.fragment_recycle_note, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        adapter.setOnPopupMenuClick(requireContext() as PopupMenuClick)
        listAdapter = view.findViewById(R.id.list_notes)
        listAdapter.adapter = adapter
        listAdapter.layoutManager = LinearLayoutManager(requireContext())
        adapter.setNotes(repository?.all as List<Note>)

        // Helper
        val helper = ItemTouchHelper(object : ItemTouchHelper.Callback() {
            override fun getMovementFlags(
                recyclerView: RecyclerView,
                viewHolder: RecyclerView.ViewHolder
            ): Int {
                val swipeFlag = ItemTouchHelper.START or ItemTouchHelper.END
                return makeMovementFlags(0, swipeFlag)
            }

            override fun onMove(
                recyclerView: RecyclerView,
                viewHolder: RecyclerView.ViewHolder,
                target: RecyclerView.ViewHolder
            ): Boolean {
                return false
            }

            override fun onSwiped(viewHolder: RecyclerView.ViewHolder, direction: Int) {
//                val position = viewHolder.adapterPosition
                val position = viewHolder.bindingAdapterPosition
                val holder = viewHolder as NoteHolder
                val note = holder.note
                note!!.id?.let { repository.delete(it) }
                adapter.delete(repository.all as List<Note>, position)
            }
        })
        helper.attachToRecyclerView(listAdapter)
    }

    fun passData(note: Note?): Note? {
        this.note = note
        return note
    }

    fun delete(note: Note, position: Int) {
        note.id?.let { repository?.delete(it) }
        adapter.delete(repository?.all as List<Note>, position)
    }

    fun update(note: Note?) {
        repository?.update(note)
        adapter.setNotes(repository?.all as List<Note>)
    }

    fun create(note: Note?) {
        repository?.create(note)
        adapter.setNotes(repository?.all as List<Note>)
    }
/*
    companion object {
        const val NOTES_SAVE = "NOTES_SAVE"
    }
    */
}